### PR TITLE
[WIP] Inclusion of Pluggable Components (.NET) SDK docs

### PR DESCRIPTION
**Pull request by philliphoff**
Signed-off-by: Phillip Hoff <phillip@orst.edu>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This includes the new Pluggable Components (.NET) SDK docs from the [`dapr-sandbox/components-dotnet-sdk`](https://github.com/dapr-sandbox/components-dotnet-sdk) repo in preparation for the preview of those SDKs alongside Dapr 1.10.

### CODE REVIEW

**Comment by philliphoff** [comment]
@hhunter-ms @greenie-msft This is a draft of changes for inclusion of docs for the new Dapr Pluggable Components .NET (and eventually other) SDKs. If you have any feedback on structure, please let me know.

File: daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+type: docs
+title: "Pluggable components SDKs"
+linkTitle: "SDKs"
+weight: 2000
+description: "Develop pluggable components in your favorite language"
+no_list: true
+is_preview: true
+---
+
+The Dapr SDKs are the easiest way for you to create Dapr Pluggable Components. Choose your favorite language and get up and running with Dapr in minutes.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md]:
```suggestion
The Dapr SDKs are the easiest way for you to create pluggable components. Choose your favorite language and start creating components in minutes.
```

File: daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+## Pluggable Components SDKs
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md]:
```suggestion
## Pluggable components SDKs
```

File: daprdocs/layouts/partials/version-banner.html [hunk 3]
```diff
@@ -13,3 +13,12 @@
     {{ end }}
   </div>
 {{ end }}
+
+<!-- Check the variable that indicates whether this is a preview doc set.
+  If yes, display a banner. -->
+  {{ if .Params.is_preview }}
+  {{ $color := "primary" }}
+  <div class="pageinfo pageinfo-{{ $color }}">
+    <p>This is documentation about a preview version of Dapr features.</p>
```
Comment (msfussell) [inline daprdocs/layouts/partials/version-banner.html]:
```suggestion
    <p>This is documentation on a preview feature.</p>
```

**Review by greenie-msft — APPROVED** [review]

**Comment by greenie-msft** [comment]
The changes look good, but it looks like the build is failing. Is that because it's trying to validate links for pages that are not merged yet?

File: daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md (hunk 1 above)
Comment (philliphoff) [reply daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md]: Generally agree, but I find "get up and running creating components" a bit awkward still.  How about:

*The Dapr SDKs are the easiest way for you to create pluggable components. Choose your favorite language and start creating components in minutes.*

**Comment by philliphoff** [comment]
> The changes look good, but it looks like the build is failing. Is that because it's trying to validate links for pages that are not merged yet?

@greenie-msft I think the build failures are likely (at least, in part) due to the need to add a new Git submodule to the .NET SDK repo's docs, and that doesn't actually exist yet because those changes themselves are also WIP.  What will likely have to happen is that the SDK-specific changes get committed first, and then this change.

File: daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+is_preview: true
+---
+
+The Dapr SDKs are the easiest way for you to create pluggable components. Choose your favorite language and start creating components in minutes.
+
+## Pluggable components SDKs
+
+| Language | Status |
+|----------|:------:|
+| [.NET]({{< ref pluggable-components-dotnet >}}) | In development |
```
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md:17]:
```suggestion
| [.NET]({{< ref dotnet-sdk-docs >}}) | In development |
```

File: daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+| [Java]({{< ref java >}}) | In development |
```
Comment (philliphoff) [inline daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md]: As the docs for Go and Java aren't available yet, should these rows be removed?

Comment (philliphoff) [reply daprdocs/content/en/developing-applications/develop-components/pluggable-components/pluggable-components-sdks/_index.md]: Removed.

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**